### PR TITLE
Master

### PR DESCRIPTION
**Pull request by nilwyh**
issue: count logic is different
for other func, no metrics coming results in 'UNDETERMINED';
but for COUNT, then its value is 0, should either be OK or ALARM.

1. Fix this logic difference.
2. Add unittest for that.

### CODE REVIEW

File: monasca/tests/microservice/test_threshold_processor.py [hunk 1]
```diff
@@ -172,6 +172,33 @@ def test_process_alarms(self):
...
+        self.assertEqual('OK', json.loads(alarms[0])['state'])
+        print (json.loads(alarms[0])['sub_alarms'][0]['current_values'])
+        metrics_list = self.util.get_metrics("metrics_count_alarm")
+        for metrics in metrics_list:
+            timestamp = json.loads(metrics)['timestamp']
+            with mock.patch.object(tu, 'utcnow_ts',
+                                   return_value=timestamp):
+                tp.process_metrics(metrics)
+        alarms = tp.process_alarms()
+        print (alarms)
```
Comment (litong01) [inline monasca/tests/microservice/test_threshold_processor.py]: can you remove these debugging statement?

File: monasca/tests/microservice/test_threshold_processor.py [hunk 2]
```diff
@@ -172,6 +172,33 @@ def test_process_alarms(self):
...
+        self.assertEqual(1, len(alarms))
+        self.assertEqual(1, len(json.loads(alarms[0])['metrics']))
+        print (json.loads(alarms[0])['sub_alarms'][0]['current_values'])
```
Comment (litong01) [inline monasca/tests/microservice/test_threshold_processor.py]: same here.

File: monasca/tests/microservice/test_threshold_processor.py [hunk 3]
```diff
@@ -172,6 +172,33 @@ def test_process_alarms(self):
...
+        tp = processor.ThresholdProcessor(ad)
+        metrics_list = self.util.get_metrics("metrics_count_ok")
+        for metrics in metrics_list:
+            timestamp = json.loads(metrics)['timestamp']
+            with mock.patch.object(tu, 'utcnow_ts',
+                                   return_value=timestamp):
+                tp.process_metrics(metrics)
+        alarms = tp.process_alarms()
+        self.assertEqual(1, len(alarms))
+        print (alarms)
```
Comment (litong01) [inline monasca/tests/microservice/test_threshold_processor.py]: remove these or use debug statement.

Comment (yihanwan) [inline monasca/tests/microservice/test_threshold_processor.py]: the pull request is updated.

I removed those prints.

At 7/27/2015 15:09, litong01 :

**Pull request merged**